### PR TITLE
sssctl: user-show command added

### PR DESCRIPTION
**Pull request by patriki01**
I added function to call ``sssctl user-show`` with additional arguments.
Waiting for your comments

### CODE REVIEW

File: sssd_test_framework/utils/sssctl.py [hunk 1]
```diff
@@ -188,6 +188,29 @@ def user_checks(self, username: str, action: str = "acct", service: str = "syste
         """
         return self.host.ssh.exec(["sssctl", "user-checks", username, "-a", action, "-s", service])
 
+    def user_show(self, user: str | None = None, sid: str | None = None, uid: str | None = None) -> SSHProcessResult:
```
Comment (pbrezina) [inline sssd_test_framework/utils/sssctl.py:191]: I would also accept uid as int.

**Review by pbrezina — CHANGES_REQUESTED** [review]

File: sssd_test_framework/utils/sssctl.py (hunk 1 above)
Comment (patriki01) [reply sssd_test_framework/utils/sssctl.py:191]: That makes sense. Probably SID can be int as well.

Comment (ikerexxe) [reply sssd_test_framework/utils/sssctl.py]: I don't think SID can be an `int` as well as it contains letters and hyphens. Example:
```
S-1-5-21-1004336348-1177238915-682003330-512
```

Comment (patriki01) [reply sssd_test_framework/utils/sssctl.py]: Oh I see. I will update it. Thanks for info

**Review by ikerexxe — DISMISSED** [review]
LGTM. Thanks for the patch!

File: sssd_test_framework/utils/sssctl.py [hunk 2]
```diff
@@ -188,6 +188,29 @@ def user_checks(self, username: str, action: str = "acct", service: str = "syste
...
+        :param sid: Search by SID, defaults to None
+        :type sid: str | None
+        :param uid: Search by user ID, defaults to None
+        :type uid: int | None
+        :return: Result of called command
+        :rtype: SSHProcessResult
+        """
+        options = []
+        if user is not None:
+            options = [user]
```
Comment (pbrezina) [inline sssd_test_framework/utils/sssctl.py:206]: `options += [user]`

File: sssd_test_framework/utils/sssctl.py [hunk 3]
```diff
@@ -188,6 +188,29 @@ def user_checks(self, username: str, action: str = "acct", service: str = "syste
...
+        if sid is not None:
+            options = options + ["-s", sid]
```
Comment (pbrezina) [inline sssd_test_framework/utils/sssctl.py:208]: `options += ...`

File: sssd_test_framework/utils/sssctl.py [hunk 4]
```diff
@@ -188,6 +188,29 @@ def user_checks(self, username: str, action: str = "acct", service: str = "syste
...
+        if uid is not None:
+            options = options + ["-u", str(uid)]
```
Comment (pbrezina) [inline sssd_test_framework/utils/sssctl.py:210]: `options += ...`

**Review by pbrezina — APPROVED** [review]

**Pull request merged**